Practical Example/IosInterviewPracticeProject/IosInterviewPracticeProject/Screens/CourseListVC.swift

```diff
@@ -0,0 +1,137 @@
+//
+//  CourseListViewController.swift
+//  IosInterviewPracticeProject
+//
+//  Created by Robert Ramirez on 10/16/20.
+//  Copyright © 2020 me.robert.ramirez. All rights reserved.
+//
+
+import UIKit
+
+class CourseListVC: UIViewController {
+  private var courseContentUrl: String?
+  private var courses = [Item]()
+  private var dataSource: UITableViewDiffableDataSource<Section, Item>?
+  private let courseManager = CourseAPIService()
+  private let timeFormatter: DateComponentsFormatter = {
+    let timeFormatter = DateComponentsFormatter()
+    timeFormatter.allowedUnits = [.hour, .minute]
+    timeFormatter.unitsStyle = .short
+    return timeFormatter
+  }()
+  @IBOutlet var courseList: UITableView!
+  @IBAction func filterCourses(_ sender: UISegmentedControl) {
+    var filteredContent = [Item]()
+
+    switch Filter(rawValue: sender.selectedSegmentIndex) {
+    case .all:
+      filteredContent = courses
+    case .article:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.article }
+    case .collection:
+      filteredContent = courses.filter { $0.attributes.contentType == ContentType.collection }
+    default:
+      filteredContent = courses
+    }
+
+    DispatchQueue.main.async {
+      self.createSnapshot(from: filteredContent.sorted { $0.attributes.releasedAt > $1.attributes.releasedAt })
+    }
+  }
+
+    override func viewDidLoad() {
+      super.viewDidLoad()
+      courseList.delegate = self
+      courseList.dataSource = dataSource
+      configureDataSource()
+      fetchContent()
```

Comment:
I general, think hard if you want to initiate requests from `viewDidLoad`. `viewDidLoad`, as the name suggests, gets called when the view _loads_, not appears. It can get called for views that won't ever appear on the screen (like a tab bar's tab etc.) and can slow down your first launch experience if you initialize multiple requests for multiple VCs that aren't on the screen.